### ISSUE TITLE
Clippy misses integer arithmetic when one operand is a reference

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust-clippy#5328]
I expected this code to trigger an "integer arithmetic" warning:
```rust
#![warn(clippy::integer_arithmetic)]

pub fn foo(x: &i32) -> i32 {
    let a = 5;
    a+x
}
```
However, [no such warning is emitted](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d93225ee13584f436b8bf8a26492691d).
